### PR TITLE
Add keyboardType parameter (type TextInputType) to TextInputSettingsTile.

### PR DESCRIPTION
**Pull request by IanDarwin**
Very convenient to be able to use the keyboard types like name, phone, etc.

### CODE REVIEW

File: lib/src/settings_screen.dart [hunk 1]
```diff
@@ -1162,6 +1165,7 @@ class TextInputSettingsTile extends StatefulWidget {
     this.obscureText = false,
     this.borderColor,
     this.errorColor,
+    this.keyboardType = TextInputType.text,
```
Comment (GAM3RG33K) [inline lib/src/settings_screen.dart:1168]: Is the default value required here?

File: lib/src/settings_screen.dart [hunk 2]
```diff
@@ -1223,6 +1227,7 @@ class _TextInputSettingsTileState extends State<TextInputSettingsTile> {
           validator: widget.enabled ? widget.validator : null,
           onSaved: widget.enabled ? (value) => _onSave(value, onChanged) : null,
           obscureText: widget.obscureText,
+          keyboardType: widget.keyboardType??TextInputType.text,
```
Comment (GAM3RG33K) [inline lib/src/settings_screen.dart:1230]: The parameter already has a default value, then why check for null value again?
& again `Is the default value required here?`

**Review by GAM3RG33K — CHANGES_REQUESTED** [review]
Excellent job on this change.

I have a few changes to request in your contribution:
-  remove the default value in both optional parameter & the where the parameter is used, passing the null value is already handled in the Flutter's TextFormField widget
-  can you change the name of the new parameter to `textInputType`? `keyboardType` is not in-appropriate but is also not consistant. [Optional change]

File: lib/src/settings_screen.dart (hunk 2 above)
Comment (IanDarwin) [reply lib/src/settings_screen.dart]: Removed, as per your suggestion. Thanks.

**Comment by IanDarwin** [comment]
> Excellent job on this change.

*blush* It's pretty trivial, but powerful :-)

> I have a few changes to request in your contribution:
> 
>     * remove the default value in both optional parameter & the where the parameter is used, passing the null value is already handled in the Flutter's TextFormField widget

Done, thanks.

>     * can you change the name of the new parameter to `textInputType`? `keyboardType` is not in-appropriate but is also not consistent. [Optional change]

The reason I called it that is just for consistency with the parameter of the underlying TextField class. I don't feel very strongly either way.

**Review by GAM3RG33K — APPROVED** [review]

**Pull request merged**